Dependency Check Failure

```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.282 s
[INFO] Finished at: 2023-06-07T02:27:57Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.sonatype.ossindex.maven:ossindex-maven-plugin:3.2.0:audit (default-cli) on project cloud-storage-extension: Detected 2 vulnerable components:
Error:    org.codehaus.janino:janino:jar:3.0.16:compile; https://ossindex.sonatype.org/component/pkg:maven/org.codehaus.janino/janino@3.0.16?utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:      * [CVE-2023-33546] CWE-400: Uncontrolled Resource Consumption ('Resource Exhaustion') (7.5); https://ossindex.sonatype.org/vulnerability/CVE-2023-33546?component-type=maven&component-name=org.codehaus.janino%2Fjanino&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:    com.google.guava:guava:jar:31.1-jre:compile; https://ossindex.sonatype.org/component/pkg:maven/com.google.guava/guava@31.1-jre?utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:      * [CVE-2023-2976] CWE-378: Creation of Temporary File With Insecure Permissions (6.2); https://ossindex.sonatype.org/vulnerability/CVE-2023-2976?component-type=maven&component-name=com.google.guava%2Fguava&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:  
Error:  Excluded vulnerabilities:
Error:    - [CVE-2019-10172] CWE-611: Improper Restriction of XML External Entity Reference ('XXE') (7.5); https://ossindex.sonatype.org/vulnerability/CVE-2019-10172?component-type=maven&component-name=org.codehaus.jackson%2Fjackson-mapper-asl&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:    - [CVE-2020-36641] CWE-611: Improper Restriction of XML External Entity Reference ('XXE') (9.8); https://ossindex.sonatype.org/vulnerability/CVE-2020-36641?component-type=maven&component-name=fr.turri%2FaXMLRPC&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:    - [CVE-2020-8908] CWE-379: Creation of Temporary File in Directory with Incorrect Permissions (6.2); https://ossindex.sonatype.org/vulnerability/CVE-2020-8908?component-type=maven&component-name=com.google.guava%2Fguava&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:    - [CVE-2023-22946] CWE-269: Improper Privilege Management (9.9); https://ossindex.sonatype.org/vulnerability/CVE-2023-22946?component-type=maven&component-name=org.apache.spark%2Fspark-core_2.13&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:    - [CVE-2017-7525] CWE-184: Incomplete Blacklist (9.8); https://ossindex.sonatype.org/vulnerability/CVE-2017-7525?component-type=maven&component-name=org.codehaus.jackson%2Fjackson-mapper-asl&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
Error:  -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
Error: Process completed with exit code 1.
```

--- a/doc/changes/changes_2.7.2.md
+++ b/doc/changes/changes_2.7.2.md
@@ -1,8 +1,10 @@
-# Cloud Storage Extension 2.7.2, released 2023-??-??
+# Cloud Storage Extension 2.7.2, released 2023-06-07
 
-Code name:
+Code name: vulnerability Fixes and Dependency Updates
 
 ## Summary
+
+With this release we update user guide. Additionally, we fixed vulnerability findings in the transitive dependencies and updated dependency versions.
 
 ## Documentation
 

--- a/doc/changes/changes_2.7.2.md
+++ b/doc/changes/changes_2.7.2.md
@@ -11,37 +11,42 @@ Code name:
 ## Security
 
 * #245: Fixed dependency check vulnerability findings
+* #247: Fixed dependency check vulnerability findings
 
 ## Dependency Updates
 
 ### Compile Dependency Updates
 
+* Updated `com.exasol:import-export-udf-common-scala_2.13:1.1.0` to `1.1.1`
 * Updated `com.exasol:parquet-io-java:2.0.1` to `2.0.3`
 * Removed `com.fasterxml.jackson.core:jackson-databind:2.14.2`
 * Removed `com.fasterxml.jackson.module:jackson-module-scala_2.13:2.14.2`
 * Updated `com.fasterxml.woodstox:woodstox-core:6.5.0` to `6.5.1`
-* Updated `com.google.protobuf:protobuf-java:3.22.2` to `3.23.0`
+* Updated `com.google.guava:guava:31.1-jre` to `32.0.0-jre`
+* Updated `com.google.protobuf:protobuf-java:3.22.2` to `3.23.2`
 * Updated `io.delta:delta-core_2.13:2.2.0` to `2.3.0`
 * Updated `io.grpc:grpc-netty:1.53.0` to `1.55.1`
-* Updated `io.netty:netty-all:4.1.90.Final` to `4.1.92.Final`
-* Updated `org.alluxio:alluxio-core-client-hdfs:2.9.2` to `294`
+* Updated `io.netty:netty-all:4.1.90.Final` to `4.1.93.Final`
+* Updated `org.alluxio:alluxio-core-client-hdfs:2.9.2` to `300`
 * Updated `org.apache.hadoop:hadoop-aws:3.3.4` to `3.3.5`
 * Updated `org.apache.hadoop:hadoop-azure-datalake:3.3.4` to `3.3.5`
 * Updated `org.apache.hadoop:hadoop-azure:3.3.4` to `3.3.5`
 * Updated `org.apache.hadoop:hadoop-common:3.3.4` to `3.3.5`
 * Updated `org.apache.hadoop:hadoop-hdfs-client:3.3.4` to `3.3.5`
 * Updated `org.apache.hadoop:hadoop-hdfs:3.3.4` to `3.3.5`
+* Updated `org.scala-lang:scala-library:2.13.10` to `2.13.11`
 * Updated `org.slf4j:jul-to-slf4j:2.0.6` to `2.0.7`
 * Updated `org.slf4j:slf4j-log4j12:2.0.6` to `2.0.7`
 
 ### Test Dependency Updates
 
-* Updated `com.dimafeng:testcontainers-scala-scalatest_2.13:0.40.12` to `0.40.15`
-* Updated `com.exasol:exasol-testcontainers:6.5.1` to `6.5.2`
+* Updated `com.dimafeng:testcontainers-scala-scalatest_2.13:0.40.12` to `0.40.16`
+* Updated `com.exasol:exasol-testcontainers:6.5.1` to `6.6.0`
 * Updated `com.exasol:hamcrest-resultset-matcher:1.5.2` to `1.6.0`
+* Updated `nl.jqno.equalsverifier:equalsverifier:3.14.1` to `3.14.2`
 * Updated `org.junit.jupiter:junit-jupiter-engine:5.9.2` to `5.9.3`
 * Updated `org.mockito:mockito-core:5.2.0` to `5.3.1`
-* Updated `org.testcontainers:localstack:1.17.6` to `1.18.0`
+* Updated `org.testcontainers:localstack:1.17.6` to `1.18.3`
 
 ### Plugin Dependency Updates
 

--- a/doc/changes/changes_2.7.2.md
+++ b/doc/changes/changes_2.7.2.md
@@ -34,7 +34,6 @@ Code name:
 * Updated `org.apache.hadoop:hadoop-common:3.3.4` to `3.3.5`
 * Updated `org.apache.hadoop:hadoop-hdfs-client:3.3.4` to `3.3.5`
 * Updated `org.apache.hadoop:hadoop-hdfs:3.3.4` to `3.3.5`
-* Updated `org.scala-lang:scala-library:2.13.10` to `2.13.11`
 * Updated `org.slf4j:jul-to-slf4j:2.0.6` to `2.0.7`
 * Updated `org.slf4j:slf4j-log4j12:2.0.6` to `2.0.7`
 

--- a/pom.xml
+++ b/pom.xml
@@ -14,7 +14,7 @@
         <relativePath>pk_generated_parent.pom</relativePath>
     </parent>
     <properties>
-        <scala.version>2.13.10</scala.version>
+        <scala.version>2.13.11</scala.version>
         <scala.compat.version>2.13</scala.compat.version>
         <hadoop.version>3.3.5</hadoop.version>
         <log4j.version>2.20.0</log4j.version>
@@ -38,7 +38,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>31.1-jre</version>
+            <version>32.0.0-jre</version>
         </dependency>
         <dependency>
             <groupId>com.fasterxml.woodstox</groupId>
@@ -53,12 +53,12 @@
         <dependency>
             <groupId>io.netty</groupId>
             <artifactId>netty-all</artifactId>
-            <version>4.1.92.Final</version>
+            <version>4.1.93.Final</version>
         </dependency>
         <dependency>
             <groupId>com.exasol</groupId>
             <artifactId>import-export-udf-common-scala_${scala.compat.version}</artifactId>
-            <version>1.1.0</version>
+            <version>1.1.1</version>
             <exclusions>
                 <exclusion>
                     <groupId>org.slf4j</groupId>
@@ -301,7 +301,7 @@
         <dependency>
             <groupId>org.alluxio</groupId>
             <artifactId>alluxio-core-client-hdfs</artifactId>
-            <version>294</version>
+            <version>300</version>
             <exclusions>
                 <exclusion>
                     <groupId>log4j</groupId>
@@ -388,7 +388,7 @@
         <dependency>
             <groupId>com.google.protobuf</groupId>
             <artifactId>protobuf-java</artifactId>
-            <version>3.23.0</version>
+            <version>3.23.2</version>
         </dependency>
         <dependency>
             <groupId>com.google.cloud.bigdataoss</groupId>
@@ -563,19 +563,19 @@
         <dependency>
             <groupId>com.dimafeng</groupId>
             <artifactId>testcontainers-scala-scalatest_${scala.compat.version}</artifactId>
-            <version>0.40.15</version>
+            <version>0.40.16</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>localstack</artifactId>
-            <version>1.18.0</version>
+            <version>1.18.3</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>com.exasol</groupId>
             <artifactId>exasol-testcontainers</artifactId>
-            <version>6.5.2</version>
+            <version>6.6.0</version>
             <scope>test</scope>
         </dependency>
         <dependency>
@@ -593,7 +593,7 @@
         <dependency>
             <groupId>nl.jqno.equalsverifier</groupId>
             <artifactId>equalsverifier</artifactId>
-            <version>3.14.1</version>
+            <version>3.14.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
@@ -772,12 +772,11 @@
                         <!-- Vulnerabilities in com.fasterxml.jackson.core:jackson-core:jar:2.14.2 -->
                         <!-- CWE-400: Uncontrolled Resource Consumption ('Resource Exhaustion') from jackson-core:2.13.4 -->
                         <exclude>sonatype-2022-6438</exclude>
-                        <!-- Vulnerabilities in com.google.guava:guava:jar:31.1-jre -->
-                        <!-- CWE-379: Creation of Temporary File in Directory with Incorrect Permissions (6.2); https://ossindex.sonatype.org/vulnerability/sonatype-2020-0926 -->
-                        <exclude>CVE-2020-8908</exclude>
-                        <!-- org.apache.spark:spark-core_2.12:jar:3.3.2: CWE-269: Improper Privilege Management (6.4); https://ossindex.sonatype.org/vulnerability/CVE-2023-22946 -->
-                        <!-- This from the spark-code_2.12 version, which we release using profiles below. -->
+                        <!-- org.apache.spark:spark-core_2.13:jar:3.3.2: CWE-269: Improper Privilege Management (6.4); https://ossindex.sonatype.org/vulnerability/CVE-2023-22946 -->
+                        <!-- This from the spark-code_2.13 version, which we release using profiles below. -->
                         <exclude>CVE-2023-22946</exclude>
+                        <!-- org.apache.spark:spark-sql_2.13.jar:3.4.0: CWE-269: Improper Privilege Management (9.9); https://ossindex.sonatype.org/vulnerability/CVE-2023-22946 -->
+                        <exclude>CVE-2023-33546</exclude>
                         <!-- Vulnerabilities in org.apache.hadoop:hadoop-hdfs-client:3.3.4 -->
                         <!-- CWE-611: Improper Restriction of XML External Entity Reference ('XXE') (8.6); https://ossindex.sonatype.org/vulnerability/sonatype-2022-5732 -->
                         <exclude>sonatype-2022-5732</exclude>

--- a/pom.xml
+++ b/pom.xml
@@ -14,7 +14,7 @@
         <relativePath>pk_generated_parent.pom</relativePath>
     </parent>
     <properties>
-        <scala.version>2.13.11</scala.version>
+        <scala.version>2.13.10</scala.version>
         <scala.compat.version>2.13</scala.compat.version>
         <hadoop.version>3.3.5</hadoop.version>
         <log4j.version>2.20.0</log4j.version>

